Frontend/src/components/Header.tsx

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable react/jsx-props-no-spreading */
+/* eslint-disable no-console */
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { useState } from 'react';
+import axios from 'axios';
+import { IHeaderProps } from '../interfaces/IHeaderProps';
+
+type FormValues = {
+  newTaskTitle: string;
+};
+
+export default function Header({ tasks, setTasks }: IHeaderProps) {
+  const [newTaskInputVal, setNewTaskInputVal] = useState('');
+  const { register, handleSubmit, reset } = useForm<FormValues>();
+  const onSubmit: SubmitHandler<FormValues> = (data) => {
+    axios
+      .post('add_task', { title: data.newTaskTitle })
+      .then((response) => {
```

Comment:
I think there is some refactoring possible here. You have implemented error caching for post on the backend. If the operation was not successful, the `taskCreated` fill be returned as false. Therefore you don't need the extra `catch()` from line #32, you could just do the else statement for the `if` that you have on line 19. Right now you're catching the errors twice for the same thing - once on the BE and once on the FE. 

If, on the BE, you returned an error instead of a valid response with `({'taskCreated': False})`, then the .catch()` on the FE would make sense, as I'd then catch the error that Be returned. But right now, the BE will never return an error. 

Also, logging the error to the console.log is not the most user friendly ever, as most people do not open it. Have you considered adding it to the page instead, eg. on top of the task lists?

Reply:
That's... a very good point on the error catching. I'll change that.

Reply:
That's... a very good point on the error catching. I'll change that.

Reply:
Done.